run-dljc.sh

```diff
@@ -100,179 +116,107 @@ fi
+    # as different repos, but forks with the same content to be skipped.
+    # TODO: consider just using hash, to skip hard forks?
+    mkdir -p "${REPO_NAME_HASH}"
 
-    pushd ${REPO_NAME}-${HASH}
+    pushd "${REPO_NAME_HASH}" || exit 5
     
-    if [ ! -d ${REPO_NAME} ]; then
-        # this environment variable prevents git from prompting for username/password
-        # if the repository no longer exists
-        GIT_TERMINAL_PROMPT=0 git clone ${REPO}
+    if [ ! -d "${REPO_NAME}" ]; then
+        # this environment variable prevents git from prompting for
+	# username/password if the repository no longer exists
+        GIT_TERMINAL_PROMPT=0 git clone "${REPO}"
     else
-        rm -rf ${REPO_NAME}/dljc-out
+        rm -rf "${REPO_NAME}/dljc-out"
     fi
 
     # if the above clone command failed for whatever reason, create an
     # empty directory so that the rest of the commands fail gracefully
     # without messing with the directory structure
-    if [ ! -d ${REPO_NAME} ]; then
-        mkdir ${REPO_NAME}
-    fi
+    mkdir -p "${REPO_NAME}"
 
-    pushd ${REPO_NAME}
+    pushd "${REPO_NAME}" || exit 5
 
-    git checkout ${HASH}
+    git checkout "${HASH}"
 
-    OWNER=`echo ${REPO} | cut -d / -f 4`
+    OWNER=$(echo "${REPO}" | cut -d / -f 4)
 
     if [ "${OWNER}" = "${USER}" ]; then
-        ORIGIN=`echo ${REPOHASH} | awk '{print $3}'`
-        git remote add unannotated ${ORIGIN}
+        ORIGIN=$(echo "${REPOHASH}" | awk '{print $3}')
+        git remote add unannotated "${ORIGIN}"
     fi
+
+    REPO_FULLPATH=$(pwd)
     
-    configure_and_exec_dljc
- 
-    popd 
-    popd
+    popd || exit 5
+
+    RESULT_LOG="${OUTDIR}-results/${REPO_NAME_HASH}-wpi.log"
+    touch "${RESULT_LOG}"
 
-    cd ${PWD}
+    "${SCRIPTDIR}/configure-and-exec-dljc.sh" -d "${REPO_FULLPATH}" -u "${USER}" -t "${TOUT}" "$@" &> "${RESULT_LOG}"
+
+    popd || exit 5
 
     # if the result is unusable, we don't need it for data analysis and we can
     # delete it right away
-    if [ ${USABLE} = "no" ]; then
-	rm -rf ${REPO_NAME}-${HASH} &
+    if [ -f "${REPO_FULLPATH}/.unusable" ]; then
+        rm -rf "${REPO_NAME_HASH}" &
+    else
+        cat "${REPO_FULLPATH}/dljc-out/wpi.log" >> "${RESULT_LOG}"
     fi
+
+    cd "${OUTDIR}" || exit 5
     
-done <${INLIST}
+done <"${INLIST}"
 
-popd
+popd || exit 5
 
-unaccounted_for=`grep -Zvl "no build file found for" ${OUTDIR}-results/*.log \
+unaccounted_for=$(grep -Zvl "no build file found for" "${OUTDIR}-results/*.log" \
     | xargs -0 grep -Zvl "dljc could not run the Checker Framework" \
     | xargs -0 grep -Zvl "dljc could not run the build successfully" \
     | xargs -0 grep -Zvl "dljc timed out for" \
-    | xargs -0 echo`
-
-javafiles=`grep -oh "\S*\.java " ${unaccounted_for}`
+    | xargs -0 echo)
 
-# echo ${javafiles}
+javafiles=$(grep -oh "\S*\.java " "${unaccounted_for}")
 
-if [ ! -f cloc-1.80.pl ]; then
+if [ ! -f "${SCRIPTDIR}/../cloc-1.80.pl" ]; then
```

Comment:
The `if` and `fi` lines can be eliminated by passing the `-nc` command-line argument to `wget`.